Update config flag function descriptions and add ped config flags enum

TheMysteriousVincent opened the pull request.

TheMysteriousVincent commented:
Note: Accidentally some CRLFs were converted to LF

LeonMrBonnie commented on client/index.d.ts:
```diff
@@ -151,6 +151,49 @@ declare module "alt-client" {
+   * 
+   * See this list for reference: https://wiki.altv.mp/wiki/GTA:Ped_Config_Flags
+   */
+  export const enum PedConfigFlagNames {
```
Please change the name to something shorter like `PedConfigFlags`

LeonMrBonnie commented on client/index.d.ts:
```diff
@@ -1906,15 +1949,19 @@ declare module "alt-client" {
    * @param flag Config flag name.
    * @param state Config flag state.
    */
   export function setConfigFlag(flag: string, state: boolean): void;
```
Use the enum you added here as a type for the `flag` param.

LeonMrBonnie replied:
The same for all the other two config flag related functions.

LeonMrBonnie commented on client/index.d.ts:
```diff
@@ -151,6 +151,49 @@ declare module "alt-client" {
 
+  /**
+   * An enum that provides mappings to ped config flag names.
+   * 
```
Remove this newline here

LeonMrBonnie replied:
Also on the other TSDoc comments you made

TheMysteriousVincent replied:
@LeonMrBonnie What's the reason for that? In general a double newline represents a line break which is intended by me.

LeonMrBonnie replied:
I meant to remove the empty line there, to conform with the rest of the documentation

TheMysteriousVincent replied:
I know, but that empty line represents a line break in a description. For example, if I write
```
/*
 * Test.
 * test
 */
```
the result would be `Test. test` when generated.

When writing two line breaks in the description, it will result in an line break in the documentation:
```
/*
 * Test.
 *
 * test
 */
```
results in
```
Test.
test
```

TheMysteriousVincent replied:
@LeonMrBonnie Its about having a fine formatted description. Nobody wants to read full text without line breaks when the text content is not coherent :)

LeonMrBonnie replied:
Ah yes you are right, mb.

Lhoerion commented:
I'm not entirely convinced that such enum like here is worth adding, when it's much likely to change every GTA update and yet it's not complete list

TheMysteriousVincent commented:
> I'm not entirely convinced that such enum like here is worth adding, when it's much likely to change every GTA update and yet it's not complete list

Do the names change? I don't think so, isn't it?

LeonMrBonnie commented:
> 
> 
> I'm not entirely convinced that such enum like here is worth adding, when it's much likely to change every GTA update and yet it's not complete list

They haven't changed before so I don't think they will ever change, I mean why would they. I think it's fine to add an enum like that even if it's not complete.

LeonMrBonnie commented on client/index.d.ts:
```diff
@@ -1906,28 +1949,34 @@ declare module "alt-client" {
-   * Returns the state of the specified config flag.
+   * Returns the state of the specified ped config flag of the local player.
+   *
+   * See this list for reference: https://wiki.altv.mp/wiki/GTA:Ped_Config_Flags
```
Put this into the `@remarks` field

TheMysteriousVincent replied:
Good idea.

FabianTerhorst commented on client/index.d.ts:
```diff
@@ -151,6 +151,49 @@ declare module "alt-client" {
+    NoPedMelee = "PED_FLAG_NO_PED_MELEE",
+    SwitchingWeapon = "PED_FLAG_SWITCHING_WEAPON",
+    Alpha = "PED_FLAG_ALPHA",
+    DisablePropKnockOff = "PED_FLAG_DISABLE_PROP_KNOCK_OFF",
```
this is the only value that works with it.
You have to remove all other values from the enum.

FabianTerhorst reviewed (changes requested):
Remove the none supported ped config flags from the enum

Lhoerion commented:
Internally it was confirmed, that only one flag works.
So if you want still to make changes to this PR:
* remove link to wiki
* remove enum
* change flag parameter type from string to that string literal

TheMysteriousVincent commented:
Alright, good to know.

Fixed that so far with two side notes:
* I renamed `PedConfigFlags` to `PedConfigFlag` for a more precise naming
* `string | PedConfigFlag` type is needed as some JS/TS (trans-)compiler do not substitute enum values when ignoring the `alt-client` module (e.g. when using bundlers).

Lhoerion commented:
Enum for this one flag isn't needed, just replace it with string literal, as only one flag is valid for this function

TheMysteriousVincent commented:
> Enum for this one flag isn't needed, just replace it with string literal, as only one flag is valid for this function

@Lhoerion I get your point, though I think it is more clever to "outsource" the flags to another datatype, because then it would be more easily to maintain (e.g. adding or removing flags).

vadzz-dev commented:
It actually not related to ped at all, just some random global flags

vadzz-dev commented:
Not sure where did you get name, but the only existing flag is `DISABLE_PED_PROP_KNOCK_OFF`

TheMysteriousVincent commented:
> Not sure where did you get name, but the only existing flag is `DISABLE_PED_PROP_KNOCK_OFF`. It globally disables knock off for all peds

@vadzz-dev sauce: https://wiki.altv.mp/wiki/GTA:Ped_Config_Flags

vadzz-dev commented:
This function is not related to that list in any way. It was made for random stuff that can't be achieved with natives

TheMysteriousVincent commented:
> This function is not related to that list in any way. It was made for random stuff that can't be achieved with natives

@vadzz-dev Then the documentation was bad all the time. I fix that in a second

Lhoerion commented:
> @vadzz-dev Then the documentation was bad all the time. I fix that in a second

It wasn't bad

Lhoerion reviewed (changes requested):
> @vadzz-dev Then the documentation was bad all the time. I fix that in a second

It wasn't bad

PS

Change single qutoes to double quotes (and flag to the one VadZz has provided)

Lhoerion reviewed (approved):
> > Enum for this one flag isn't needed, just replace it with string literal, as only one flag is valid for this function
> 
> @Lhoerion I get your point, though I think it is more clever to "outsource" the flags to another datatype, because then it would be more easily to maintain (e.g. adding or removing flags).

Yes, we'd do that for any other type, but in this case it'd be nonsense, since I don't think any other flag will be added in the nearest future

Thanks for contribution, appreciated
